src/XMakeBuildEngine/Evaluation/LazyItemEvaluator.RemoveOperation.cs

```diff
@@ -24,16 +28,7 @@ protected override void SaveItems(ICollection<I> items, ImmutableList<ItemData>.
 
             public ImmutableHashSet<string>.Builder GetRemovedGlobs()
             {
-                var ret = ImmutableHashSet.CreateBuilder<string>();
-                foreach (var operation in _operations)
-                {
-                    if (operation.Item1 == ItemOperationType.Glob)
-                    {
-                        ret.Add((string)operation.Item2);
-                    }
-                }
-
-                return ret;
+                return _itemSpec.Fragments.OfType<GlobFragment>().Select(g => g.ItemSpecFragment).ToImmutableHashSet().ToBuilder();
```

Comment:
Seems inefficient to seal this into an immutable set, then get a builder out. Can you `return new ImmutableHashSet<string>.Builder(_itemSpec.Fragments.OfType<GlobFragment>().Select(g => g.ItemSpecFragment))`  instead?